python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -380,3 +381,89 @@ def close(self) -> None:
         if self.is_local_node(self._reader_node_id):
             self.ensure_registered_as_reader()
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
+
+
+@PublicAPI(stability="alpha")
+class MultiChannel(ChannelInterface):
+    def __init__(
+        self,
+        writer: Optional[ray.actor.ActorHandle],
+        readers: List[Optional[ray.actor.ActorHandle]],
+        _local_channel: Optional[LocalChannel] = None,
+        _remote_channel: Optional[Channel] = None,
+    ):
+        """
+        Can be used to send data to different readers via different channels.
+        For example, if the reader is in the same worker process as the writer,
+        the data can be sent via LocalChannel. If the reader is in a different
+        worker process, the data can be sent via shared memory channel.
```

Comment:
From the current documentation it is not quite clear when we should create a `MultiChannel`, and what are the expected args. Please add doc for args as well.

Reply:
Updated https://github.com/ray-project/ray/pull/45591/commits/af458bff73a2ef8a4972fa013e280b2662fcc5d9.